lib/vector-config-macros/src/ast/field.rs

```diff
@@ -89,6 +90,15 @@ impl<'a> Field<'a> {
     }
 }
 
+impl<'a> Spanned for Field<'a> {
+    fn span(&self) -> proc_macro2::Span {
+        match self.original.ident.as_ref() {
+            Some(ident) => ident.span(),
+            None => self.original.ty.span(),
+        }
+    }
+}
+
```

Comment:
This change is us trying to generate better compiler error output from the procedural macro, specifically: for named fields (`foo: String`) we underline the field name when the macro encounters an error, and if it's a tuple field (the string in `struct OuterStruct(String)`), we underline the type itself.

Prior to this, the spanning (how the compiler tracks tokens for the purposes of ensuring errors end up underlining the right code in error output) wasn't up to snuff so errors weren't being attached as narrowly as we could actually achieve.